src/gui/socketapi.cpp

```diff
@@ -617,8 +749,20 @@ class GetOrCreatePublicLinkShare : public QObject
 
 #endif
 
-void SocketApi::command_COPY_PUBLIC_LINK(const QString &localFile, SocketListener *)
+void SocketApi::command_COPY_PUBLIC_LINK(const QString &localFileC, SocketListener *)
 {
+    QString localFile = localFileC;
+
+#if defined(Q_OS_WIN)
+    OCC::ConfigFile Cfg;
+    char letter[2];
+    letter[0] = localFile.toStdString().c_str()[0];
+    letter[1] = 0;
+
+    if (!QString(letter).compare(Cfg.defaultFileStreamLetterDrive().toUpper()))
+        localFile.replace(0, 3, QStandardPaths::writableLocation(QStandardPaths::AppDataLocation) + "/cachedFiles/");
+#endif
```

Comment:
Can't this be moved into a seperate sanitizer step within localFile processing (because this ifdef appears several times in similar form)?